sources/input/error.c

```diff
@@ -6,36 +6,56 @@
 /*   By: lfarias- <lfarias-@student.42.rio>         +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/03/07 15:14:37 by lfarias-          #+#    #+#             */
-/*   Updated: 2023/03/14 13:27:15 by lfarias-         ###   ########.fr       */
+/*   Updated: 2023/04/10 19:09:20 by lfarias-         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "../../headers/mini_rt.h"
 #include <errno.h>
 #include <string.h>
 
-void	print_correct_usage(void)
+char	*get_error_msg(int error_code)
 {
-	ft_putendl_fd("USAGE: ./miniRT <filename>.rt", STDERR_FILENO);
+	char	*error_msg[12];
+
+	error_msg[err_usage] = "USAGE: ./miniRT <filename>.rt";
+	error_msg[err_file_type] = "the current file is not a .rt file";
+	error_msg[err_file_fmt] = "the file formatting is invalid";
+	error_msg[err_file_perm] = "could not read file: permission denied";
+	error_msg[err_file_not_found] = "file not found";
+	error_msg[err_obj_attr_fmt] = "invalid format for object's attribute";
+	error_msg[err_obj_attr_value] = "invalid value for object's attribute";
+	error_msg[err_obj_attr_miss] = "wrong attr size in object declaration";
+	error_msg[err_obj_dup] = "multiple unique objects [light, camera, a_light]";
+	error_msg[err_obj_invalid] = "invalid object";
+	return (error_msg[error_code]);
```

Comment:
About that, I did some research and this is a valid way to return string literals

https://www.geeksforgeeks.org/storage-for-strings-in-c/

Usually string literals are stored in the data segment of the program's memory, apart from the stack and heap segments.

If we do

`char name[] = "Leo";
and then return (name)`
then that will get us in trouble because name will be created on the stack like any other local/auto variable because it's an array.

But 

`char *name = "Leo";
return (name); `
Is ok because this is a pointer.

I also went to ChatGPT and got the same answer:

> In the C programming language, string literals like "Leonardo" are typically stored in a read-only section of memory, such as the program's code segment or a read-only data segment.
> When you declare a character pointer variable like char *name = "Leonardo";, the compiler will typically allocate a block of memory to hold the string literal "Leonardo", and then initialize the pointer variable name to point to the first character of that memory block. The memory block containing the string literal is usually stored in a read-only section of memory to prevent accidental modifications of the string.
> The exact location where the string literal is stored may depend on the compiler, operating system, and platform you are using. However, it is typically stored in a memory location that is separate from the stack and heap, which are used for storing local variables and dynamically allocated memory, respectively.

But, I do think we could just strdup those strings when returning them and free after their use just to be extra cautious.